src/io/http/src/test/scala/services/SearchWriterSuite.scala

```diff
@@ -0,0 +1,225 @@
+      "indexJson" -> json))
+
+    retryWithBackoff() {
+      if (getExisting(azureSearchKey, testServiceName).contains(in)) {
+        AzureSearchWriter.write(df2,
+          Map("subscriptionKey" -> azureSearchKey,
+            "actionCol" -> "searchAction",
+            "serviceName" -> testServiceName,
+            "indexJson" -> json))
+      }
+    }
+  }
+
+  test("push a large number of docs to an index") {
+    val numDocs = 100000
+    val df = createTestData(numDocs)
+    val in = generateIndexName()
+    val json = createSimpleIndexJson(in)
+
+    AzureSearchWriter.write(df,
+      Map("subscriptionKey" -> azureSearchKey,
+        "actionCol" -> "searchAction",
+        "serviceName" -> testServiceName,
+        "indexJson" -> json))
+  }
+
+  test("push docs with custom batch size") {
+    val numDocs = 100000
+    val df = createTestData(numDocs)
+    val in = generateIndexName()
+    val json = createSimpleIndexJson(in)
+
+    AzureSearchWriter.write(df,
+      Map("subscriptionKey" -> azureSearchKey,
+        "actionCol" -> "searchAction",
+        "serviceName" -> testServiceName,
+        "indexJson" -> json,
+        "batchSize" -> "1000"))
+  }
+
+  test("throw useful error when given badly formatted json") {
+    val in = generateIndexName()
+    val badJson = s"""
+       |{
+       |    "name": "$in",
+       |    "fields": [
+       |      {
+       |        "name": "id",
+       |        "type": "Edm.String",
+       |        "key": true,
+       |        "facetable": false
+       |      },
+       |    {
+       |      "name": "someCollection",
+       |      "type": "Collection(Edm.String)",
+       |      "searchable": false,
+       |      "sortable": true,
+       |      "facetable": false
+       |    },
+       |    {
+       |      "name": "text",
+       |      "type": "Edm.String",
+       |      "filterable": false,
+       |      "sortable": false,
+       |      "facetable": false
+       |    }
+       |    ]
+       |  }
+    """.stripMargin
+    assertThrows[IllegalArgumentException] {
+      SearchIndex.createIfNoneExists(azureSearchKey, testServiceName, badJson)
+    }
+  }
+
+  test("throw useful error when given mismatched schema and document fields") {
+    val docs = for (i <- 0 until 4)
+      yield ("upload", s"$i", s"file$i", s"text$i")
+    val mismatchDF = docs.seq.toDF("searchAction", "badkeyname", "fileName", "text")
+    val in = generateIndexName()
+    val json = createSimpleIndexJson(in)
+
+    assertThrows[IllegalArgumentException] {
+      AzureSearchWriter.write(mismatchDF,
+        Map("subscriptionKey" -> azureSearchKey,
+          "actionCol" -> "searchAction",
+          "serviceName" -> testServiceName,
+          "indexJson" -> json))
+    }
+
+  }
+
+  test("run asserts") {
+    runAsserts(Map(
+      "test-0" -> 4,
+      "test-1" -> 10,
+      "test-2" -> 100000))
+  }
+
+//  // only use when you're done with testing/need to clear all of the indices within the service
```

Comment:
we should probably clean up after this testsuite but do it in a more surgical manner
aka create an index then delete it if its been created

Reply:
You can do this with beforeAll and afterAll functions